blog/writing-reusable-julia-code.md

```diff
@@ -0,0 +1,241 @@
+---
+title: Writing Reusable Code in Julia
+author: Carlos Paniagua
+people:
+  - Carlos Paniagua
+date: 2024-03-25
+slug: writing-reusable-julia-code
+team: GSDC (Graphics, Software and Data Core)
+description: Best practices on writing reusable software in the Julia ecosystem.
+tags:
+  - Julia
+  - reusability
+
+---
+
+# Writing Reusable Code in Julia - Carlos Paniagua
+
+### Use case
+
+-   You are working on a research project that requires computation. 💻👩‍💻
+
+-   You write your code; you test it (👍) and it just works! 🚀
+
+-   You decide your code is general enough that it could be useful for solving related problems. 🤓
+
+-   You add some documentation and host your code in some public repository and tell other people about it. 🎤
+
+-   Other people realize your code could be useful to solve their problems! 😎
+
+-   They take your code, try to use it and... it doesn't work 😱
+
+-   If they're lucky, the code "might work" but the results are not the expected ones 😱😱
+
+This is a very common problem in the scientific research community. Here we present an introduction about best practices to building reusable code in the Julia ecosystem.
+
+## Our definition of reusable code
+
+[Reusable code](https://en.wikipedia.org/wiki/Reusability) is code that is easy to distribute and use (correctly and consistently).
+
+<div style="background-color: #f7f7f7; border-left: 6px solid #007bff; padding: 10px; margin-bottom: 10px;">
+  <strong>Note:</strong> If you'd like to follow along with the rest of the discussion, you'll need Git, a text editor or IDE (VS Code recommended with the Julia extension) and Julia (naturally).
+</div>
+
+To write reusable Julia code we need to understand how to *manage dependencies* in Julia.
+
```

Comment:
Before getting into everything, Perhaps explain what the example project that is being built?

Reply:
I think even explaining that we're going to use a toy project that prints a math fact as our basis will help ground the reader. At this point of the blog post the reader is getting started without any idea as to what they are going

Reply:
Thanks for the perspective. My intention, as I discuss at the top, is to provide a short walkthrough of a few best practices for writing reusable code. Namely, the post discusses (with examples) dependency management, how to package code, a few suggestions for configuring a development environment, and publishing the project to a remote repository where other people can get and use the code. I'd like to keep the spotlight on this.